core/src/main/scala/kafka/log/Log.scala

```diff
@@ -156,7 +157,11 @@ class Log(@volatile var dir: File,
   /* Load the log segments from the log files on disk */
   private def loadSegments() {
     // create the log directory if it doesn't exist
-    dir.mkdirs()
+    try {
+      Files.createDirectory(dir.toPath)
+    } catch {
+      case e: FileAlreadyExistsException => None
+    }
```

Comment:
None is returned to no val, I think this is unnecessary. What do you want to do if directory already exist? Do you want to log in somewhere? Or maybe just ignore it?